crates/client/assets/schema.sdl

```diff
@@ -335,6 +335,10 @@ input ExcludeInput {
 	messages: [Nonce!]!
 }
 
+type FailureDuringBlockProductionStatus {
+	blockHeight: U32!
+}
+
```

Comment:
What can lead to a transaction failing with a `FailureDuringBlockProductionStatus`? 

I’ve noticed this status doesn’t include a `reason` property like `FailureStatus` does.

Reply:
> What can lead to a transaction failing with a FailureDuringBlockProductionStatus?

There could be a change in state (for example, different gas price) between these two points in time:
1. executor grabs the transaction from the TxPool (where it sits, validated and ready to be executed)
2. executor actually tries to execute the transaction

> I’ve noticed this status doesn’t include a reason property like FailureStatus does.

The exact shape of the API is not finalized yet. We decided to add the `reason` to `FailureDuringBlockProductionStatus` as well. There will be more commits added to this PR.

Reply:
`reason` is now added to the `FailureDuringBlockProductionStatus` variant